pkg/resolver/search.go

```diff
@@ -295,10 +305,30 @@ func formatDataMap(data map[string]interface{}) map[string]interface{} {
 			item[key] = strconv.FormatInt(int64(v), 10)
 		case map[string]interface{}:
 			item[key] = formatLabels(v)
+		case []interface{}:
+			item[key] = formatArray(v)
 		default:
 			klog.Warningf("Error formatting property with key: %+v  type: %+v\n", key, reflect.TypeOf(v))
 			continue
 		}
 	}
 	return item
 }
+
+func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
+	var whereDs []exp.Expression
+	for _, filter := range input.Filters {
+		var values []string
+		if len(filter.Values) > 0 {
+			for _, val := range filter.Values {
+				values = append(values, *val)
+			}
+		}
+		if filter.Property == "cluster" {
+			whereDs = append(whereDs, goqu.C(filter.Property).In(values).Expression())
+		} else {
+			whereDs = append(whereDs, goqu.L(`"data"->>?`, filter.Property).In(values).Expression())
+		}
```

Comment:
Couple changes:
- Check if a filter has no values.
- Allocate array instead of append()

```suggestion
		if len(filter.Values) > 0 {
			values := make([]string, len(filter.Values))
			for i, val := range filter.Values {
				values[i] = *val
			}
		
			if filter.Property == "cluster" {
				whereDs = append(whereDs, goqu.C(filter.Property).In(values).Expression())
			} else {
				whereDs = append(whereDs, goqu.L(`"data"->>?`, filter.Property).In(values).Expression())
			}
		} else {
			klog.Warnf("Ignoring filter [%s] because it has no values", filter.Property)
		}
```

Reply:
The graphQL schema should also enforce that values isn't empty.  We can add `!` here.

https://github.com/stolostron/search-v2-api/blob/main/graph/schema.graphqls#L27

Reply:
Done